api/intents_api.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TypedDict
+
+from flask import Response, render_template
+
+from api import converters
+from chromestatus_openapi.models import PostIntentRequest, MessageResponse
+from framework import basehandlers
+from framework import cloud_tasks_helpers
+from framework import permissions
+from internals import processes
+from internals import stage_helpers
+from internals.core_enums import INTENT_STAGES_BY_STAGE_TYPE
+from internals.core_models import FeatureEntry, Stage
+from internals.review_models import Gate
+from pages.intentpreview import compute_subject_prefix
+
+
+class IntentOptions(TypedDict):
```

Comment:
Agree with the earlier comment that we should use openapi for the tasks API. It would prevent us from having to maintain this manually. Can you add a comment about the location of the task code that consumes these parameters?

Reply:
Added a comment noting what this was being used for. dicts with varying properties are sent to  the `enqueue_task()` helper - I'd be curious to know what is the best way to document this in OpenAPI, since the props are so variable. Make an OpenAPI type definition for each possible dict structure passed to `enqueue_task()`?

Reply:
Could you give me an example of how it varies?

If it can be complex but they all share similar fields, you could use composition via `allOf` and `discriminator` More about it here: https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/ (Check the `allOf` section)

Reply:
Most of the invocations of enqueue_task() pass in a dict for `task_params` that have various key-value pairs that are unique fields for populating the email task, e.g. this [invocation in feature_links.py](https://github.com/GoogleChrome/chromium-dashboard/blob/main/internals/feature_links.py#L253-L257) uses completely different keys.

However, I see what you're saying on how we could handle this. Perhaps this is just a matter of making new schemas for each possible task_param structure and using `anyOf`. Like this example from the link:
```yaml
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              anyOf:
                - $ref: '#/components/schemas/PetByAge'
                - $ref: '#/components/schemas/PetByType'
```